scripts/deploy

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Commits all changed files and pushes to GitHub target branch.
+set -e # Exit with nonzero exit code if anything fails
+
+cd "$(dirname "$0")/.."
+# Pull requests and commits to other branches cannot commit and push.
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+    echo "Skipping deploy"
+    exit 0
+fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Decrypt the deploy key.
+openssl aes-256-cbc -K $encrypted_f45bcf07afa7_key \
+  -iv $encrypted_f45bcf07afa7_iv -in imjoy_github_deploy.enc \
+  -out imjoy_github_deploy -d
+
+# Set the correct permission
+chmod 600 ./imjoy_github_deploy
+
+# Import the deploy key
+eval "$(ssh-agent -s)"
+ssh-add ./imjoy_github_deploy
+
+# Clone the existing gh-pages for this repo into gh-pages/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deploy).
+git clone $REPO gh-pages
+cd gh-pages
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+git config user.name "Travis CI"
+git config user.email "travis@travis-ci.org"
+
+# Copy data we're interested in to gh-pages directory
+# UPDATE THIS!
+# cp -Rf $HOME/dist/* .
+# Copy dirs and files and that we want to update.
+cp -Rf ../docs/* ./docs/
```

Comment:
And copy all files in `master` root `/docs` to `gh-pages/docs` where `gh-pages/` is the root of the deploy `gh-pages` branch (repo).

Reply:
We're only copying `master/docs` to `gh-pages/docs` but we are copying to a directory where we have cloned the repo and checked out a new orphan branch `gh-pages`, so that will contain all the contents of master including `/web`.

We could create the `gh-pages` branch manually first and only commit the docs files outside of the deploy script. Then when the script would run, it would just update the docs files.

I'll change it to how you suggest.

Reply:
Great, thanks. I think one reason is that Github pages seems only support serving from a `/docs` folder in master branch, then the other choice is serve the entire `gh-pages` branch ([ref](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/)). That would require to remove the `/docs` folder from `gh-pages` and take out all the files to the root of `gh-pages`.

Reply:
Good! I'll change to that dir structure.